core/chain_manager.go

```diff
@@ -616,14 +615,12 @@ func (self *ChainManager) InsertChain(chain types.Blocks) (int, error) {
 		stats      struct{ queued, processed, ignored int }
 		tstart     = time.Now()
 
-		nonceDone    = make(chan nonceResult, len(chain))
-		nonceQuit    = make(chan struct{})
 		nonceChecked = make([]bool, len(chain))
 	)
 
 	// Start the parallel nonce verifier.
-	go verifyNonces(self.pow, chain, nonceQuit, nonceDone)
-	defer close(nonceQuit)
+	nonceAbort, nonceResults := verifyBlockNonces(self.pow, chain)
+	defer close(nonceAbort)
```

Comment:
Is it really possible to have subsequent writes with ```defer close(nonceResults)``` here? Give than it blocks below, waiting for all nonce results.

Also, the deferred close of ```abort``` channel will cause the spawned go routine that feeds ```tasks``` to return, which in turn closes ```tasks``` channel. If there are any writers left writing after that, they would panic on reading the ```tasks``` chan, right?

So closing nonceResults after closing abort should be fine?